### PR TITLE
githooks: delete vendor and pkg/ui/yarn-vendor dirs if unneeded 

### PR DESCRIPTION
**Pull request by rafiss**
Epic: None
Release note: None

### CODE REVIEW

File: githooks/post-checkout [hunk 1]
```diff
@@ -5,3 +5,12 @@ if [ "$1" != "$2" ]; then # previous ref != new ref.
   git clean -fd "pkg/**.pb.go" "pkg/**.pb.gw.go" "pkg/**.eg.go" "pkg/**.og.go" "pkg/**generated.go"
   exec git submodule update --init --recursive
 fi
+
+# Only if this is a branch checkout.
+if [ "$3" == "1" ]; then 
+    BRANCH_NAME=$(git symbolic-ref --short -q HEAD)
+    if [ ${BRANCH_NAME} == "master" ]; then
+        git clean -fdx "vendor"
```
Comment (rickystewart) [inline githooks/post-checkout:13]: Nit: I think `rm -rf vendor` accomplishes more or less the same thing here.

**Review by rickystewart — APPROVED** [review]

File: githooks/post-checkout (hunk 1 above)
Comment (rafiss) [reply githooks/post-checkout:13]: that's dependent on being in the root directory of the repo, whereas i believe `git clean [pattern]` does not

**Comment by rafiss** [comment]
the failure was an unrelated flake in `TestLeaseTransferRejectedIfTargetNeedsSnapshot/reject-after-revoke=true` (https://github.com/cockroachdb/cockroach/issues/92192)

**Comment by rafiss** [comment]
tftr!

bors r=rickystewart

**Comment by rafiss** [comment]
i think i'm wrong 🤦 

bors r-

**Comment by rafiss** [comment]
i've fixed this up and also made it delete the pkg/ui/yarn-vendor directory if it's not needed.

**Pull request merged**